src/Compilers/Core/Portable/Diagnostic/Diagnostic_SimpleDiagnostic.cs

```diff
@@ -99,7 +99,16 @@ public override string GetMessage(IFormatProvider formatProvider = null)
                 }
 
                 var localizedMessageFormat = _descriptor.MessageFormat.ToString(formatProvider);
-                return string.Format(formatProvider, localizedMessageFormat, _messageArgs);
+
+                try
+                {
+                    return string.Format(formatProvider, localizedMessageFormat, _messageArgs);
+                }
+                catch (Exception)
+                {
+                    // Analyzer reported diagnostic with invalid format arguments, so just return the unformatted message.
+                    return localizedMessageFormat;
```

Comment:
Probably, but that needs a bit of additional work as GetMessage is not invoked by the analyzer driver, but by the analyzer host (command line compiler, error list and IDE diagnostic service, etc.) and so each host would need to setup a handler for the exception and take action. I'd let it be as is until we get any customer request for it.